File: server/src/main/java/de/pnp/manager/network/serializer/CurrencyDeserializer.java
package de.pnp.manager.network.serializer;

import com.fasterxml.jackson.core.JsonParser;
import com.fasterxml.jackson.databind.DeserializationContext;
import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
import de.pnp.manager.model.Currency;

import java.io.IOException;

public class CurrencyDeserializer extends StdDeserializer<Currency> {

    public CurrencyDeserializer() {
        super(Currency.class);
    }

    @Override
    public Currency deserialize(JsonParser p, DeserializationContext ctxt) throws IOException {
        return new Currency(p.getValueAsString());
    }
}

File: app/src/main/java/de/pnp/manager/app/network/serialization/CurrencyDeserializer.kt
package de.pnp.manager.app.network.serialization

import com.fasterxml.jackson.core.JsonParser
import com.fasterxml.jackson.databind.DeserializationContext
import com.fasterxml.jackson.databind.deser.std.StdDeserializer
import de.pnp.manager.app.model.Currency
import de.pnp.manager.model.ICurrency

class CurrencyDeserializer : StdDeserializer<ICurrency>(ICurrency::class.java) {

    override fun deserialize(
        p: JsonParser,
        ctxt: DeserializationContext
    ): Currency {
        return Currency(p.valueAsString)
    }
}